### PR TITLE
Add a desktop shortcut toggle to a new page on the installer

### PR DESCRIPTION
**Pull request by ghost**
This PR adds a new page to the installation setup which allows toggling the creation of the Desktop Shortcut
I could have added the start menu one, but I feel like most people would want that, unlike the desktop one

Preview: (Is difficult to crop screenshot admin apps, so sorry for 2006 YT quality) 
image

### CODE REVIEW

**Comment by kleidis** [comment]
The CI don't build the installer on Pr, but I tested on my fork and it did build fine

File: src/installer/lime3ds.nsi [hunk 1]
```diff
@@ -7,11 +7,6 @@
 ;   probably also want vscode extension: https://marketplace.visualstudio.com/items?itemName=idleberg.nsis
 ;   makensis /DPRODUCT_VERSION=<release-name> /DPRODUCT_VARIANT=<msvc/msys2> <this-script>
 
-; Require /DPRODUCT_VERSION=<release-name> to makensis.
-!ifndef PRODUCT_VERSION
-  !error "PRODUCT_VERSION must be defined"
-!endif
-
```
Comment (OpenSauce04) [inline src/installer/lime3ds.nsi]: This is not duplicate code. The values being checked for between the two code blocks are different.

File: src/installer/lime3ds.nsi [hunk 2]
```diff
@@ -118,6 +121,25 @@ Var DisplayName
   ${EndIf}
 !macroend
 
+Function DesktopShortcutPageCreate
```
Comment (OpenSauce04) [inline src/installer/lime3ds.nsi]: Please use camelCase here.

File: src/installer/lime3ds.nsi [hunk 3]
```diff
@@ -118,6 +121,25 @@ Var DisplayName
...
+  !insertmacro MUI_HEADER_TEXT "Create Desktop Shortcut" "Would you like to create a desktop shortcut?"
+  nsDialogs::Create 1018
+  Pop $DesktopShortcutPageDialog
+  ${If} $DesktopShortcutPageDialog == error
+    Abort
+  ${EndIf}
+
+  ${NSD_CreateCheckbox} 0u 0u 100% 12u "Create a desktop shortcut"
+  Pop $DesktopShortcutCheckbox
+  ${NSD_Check} $DesktopShortcutCheckbox
```
Comment (OpenSauce04) [inline src/installer/lime3ds.nsi]: The checkbox's state is not set here, so every time you return to the page, the checkbox is re-checked even if you previously deselected it. Please review the documentation for `NSD_SetState`.

**Review by OpenSauce04 — CHANGES_REQUESTED** [review]

File: src/installer/lime3ds.nsi (hunk 3 above)
Comment (ghost) [reply src/installer/lime3ds.nsi]: Should checkbox be on b default?

**Review by OpenSauce04 — APPROVED** [review]
I tested several different combinations of actions in the installer, and this code behaves as expected. Thanks for your contribution.

**Pull request merged**